### PR TITLE
Learning Path 📚

### PR DESCRIPTION
**Pull request by Skn0tt**
As discussed in a previous Blitz Meetup, a learning path is a great resource to point novice & new developers towards.

### CODE REVIEW

**Comment by Skn0tt** [comment]
This PR is barely more than a small draft, y'all are welcome to suggest content you'd like to see included :)

**Comment by muco-rolle** [comment]
More content on data modeling in SQL will be great as many JS developers are used to NoSQL (MongoDB) database

**Comment by JuanM04** [comment]
It could contain a mention to [Intro to React](https://reactjs.org/tutorial/tutorial.html) and something that explains how the Backend works for frontend developers (like [Backend 101 for Front-End Developers](https://www.youtube.com/watch?v=fbDoO83t1eM))

File: app/pages/docs/learning-path.mdx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+- TODO: writing code to be easily testable
+- TODO: cypress guide
+
+## Software Architecture {#architecture}
+
+As your application grows bigger, some structure may be neccessary.
+There's a million ways on how to structure your project, here's some good
+guidance:
+
+- TODO: a good guide on onion / clearn architecture
```
Comment (flybayer) [inline app/pages/docs/learning-path.mdx:35]: Honestly I'm not sure this should be in our normal learning path. This is more like a path for advanced users who want to explore advanced architecture. Having it here is probably too intimidating for most people.

File: app/pages/docs/learning-path.mdx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+This document is inspired by the
+[Frontend Developer Roadmap](https://roadmap.sh/frontend). It curates
+material on concepts & ideas, tools & libraries and history & trivia
+useful to become better at building applications. It is tailored to Blitz,
+but most will easily apply elsewhere.
+
+## todo: missing categories {#missing}
+
+TODO: which categories are missing?
```
Comment (flybayer) [inline app/pages/docs/learning-path.mdx:17]: - Database modeling
- How to do multi-tenant
- Background processing, cron jobs
- Authorization strategies
- Emails
- Billing integrations (Stripe, Paddle, etc.)

**Review by flybayer — COMMENTED** [review]
Good idea!!

**Comment by Skn0tt** [comment]
> Intro to React

@JuanM04 great idea! The official "Intro to React" uses class-based components though, which I don't think is a good fit. Do you know of a good tutorial that uses Function Components / Hooks?

**Comment by JuanM04** [comment]
@Skn0tt I've found two resources from freeCodeCamp:

- a complete ten-hour tutorial: [_Full React Course 2020 - Learn Fundamentals, Hooks, Context API, React Router, Custom Hooks_](https://www.youtube.com/watch?v=4UZrsTqkcW4);
- and this [_React.js Handbook for Front End Developers_](https://www.freecodecamp.org/news/react-beginner-handbook/).

**Comment by Skn0tt** [comment]
Okay so now there's a bit of content in here, but not a lot. Should we launch this and ask people to add more stuff, or wait until this is more complete?

**Comment by jackbravo** [comment]
As an alternative to the official intro to react, I enjoyed this free egghead introduction to react: https://egghead.io/courses/the-beginner-s-guide-to-react

**Comment by jackbravo** [comment]
And also probably we should link to the official prisma documentation? But I'm not sure what would be the best entry place. Maybe the [What is Prisma](https://www.prisma.io/docs/concepts/overview/what-is-prisma) page?

**Comment by flybayer** [comment]
@Skn0tt yeah let's ship it. Easier for folks to add to it. Maybe we can improve the todo items with a comment like (want to contribute this part?)

**Comment by Skn0tt** [comment]
Let's ship it then!

File: app/pages/docs/learning-path.mdx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+material on concepts & ideas, tools & libraries and history & trivia
+useful to become better at building applications. It is tailored to Blitz,
+but most will easily apply elsewhere.
+
+<!-- prettier-ignore-start -->
+:::info
+Links marked with 🥷 are advanced. If you're dipping your toes into
+web development, it's totally OK to skip them.
+:::
+<!-- prettier-ignore-end -->
```
Comment (JuanM04) [inline app/pages/docs/learning-path.mdx:20]: Looks nice!
Sync your branch with `main` and use the new `Card` component

```suggestion
<Card type="info">

Links marked with 🥷 are advanced. If you're dipping your toes into
web development, it's totally OK to skip them.

</Card>
```

**Comment by flybayer** [comment]
Thanks @Skn0tt! Looks good except for the lint errors

**Comment by Skn0tt** [comment]
Fixed the lint errors.

**Pull request merged**